mllib/src/main/scala/org/apache/spark/ml/tree/impl/BaggedPoint.scala

```diff
@@ -60,20 +68,24 @@ private[spark] object BaggedPoint {
       subsamplingRate: Double,
       numSubsamples: Int,
       withReplacement: Boolean,
+      extractSampleWeight: (Datum => Double) = (_: Datum) => 1.0,
```

Comment:
Just checking my understanding here, but is the intention to in future support something like `WeightedTreePoint` (or amend `TreePoint` to include a weight), which is constructed in turn from `Instance` rather than `LabeledPoint`, and then the function passed can be `... => point.weight` or similar?